### PR TITLE
Make copy_from return an ImageResult instead of a boolean

### LINKED ISSUE #901: [API design] copy_from returns bool
Hello, and thank you for this very useful crate!
  I stumbled upon [`GenericImage::copy_from`](https://docs.rs/image/0.21.1/image/trait.GenericImage.html#method.copy_from), that returns `false` when the operation fails. This is not very idiomatic in rust, and makes error handling more difficult. It would be nice if the method could return a `Result<()>`. This would prevent crate users from inadvertently ignoring an error.

### PR DESCRIPTION
**Pull request by lovasoa**
Fixes #901

### CODE REVIEW

File: src/image.rs [hunk 1]
```diff
@@ -618,16 +618,15 @@ pub trait GenericImage: GenericImageView {
     /// In order to copy only a piece of the other image, use `sub_image`.
     ///
     /// # Returns
-    /// `true` if the copy was successful, `false` if the image could not
-    /// be copied due to size constraints.
+    /// Returns an error if the image is too large to be copied at the given position
     fn copy_from<O>(&mut self, other: &O, x: u32, y: u32) -> bool
```
Comment (fintelia) [inline src/image.rs]: The `bool` here should actually be `ImageResult<()>`...

Comment (lovasoa) [reply src/image.rs]: Wow ! I messed this up when reopening the old PR.

**Comment by fintelia** [comment]
Thanks!

**Comment by lovasoa** [comment]
@fintelia Thanks to you for merging this

**Pull request merged**